src/discord.re

```diff
@@ -1,29 +1,31 @@
-type client;
-
-type message;
-
-type user;
-
-type channel;
-
-external createClient : unit => client = "Client" [@@bs.module "discord.js"] [@@bs.new];
-
-external login : client => string => unit = "" [@@bs.send];
-
-external onReady : client => _ [@bs.as "ready"] => (unit => unit) => unit = "on" [@@bs.send];
-
-external onMessage : client => _ [@bs.as "message"] => (message => unit) => unit =
-  "on" [@@bs.send];
-
-external getContent : message => string = "content" [@@bs.get];
-
-external getChannel : message => channel = "channel" [@@bs.get];
-
-external getType : channel => string = "type" [@@bs.get];
-
-external getAuthor : message => option user =
-  "author" [@@bs.get] [@@bs.return null_undefined_to_opt];
-
-external isBot : user => bool = "bot" [@@bs.get];
-
-external reply : message => string => unit = "" [@@bs.send];
+module Channel = {
+  type t;
+  /* May change later to TextChanngel.t, VoiceChannel.t, and so on */
+  type cType =
+    | Text
+    | Voice
+    | DM
+    | Group;
+  external channelType : t => cType = "type" [@@bs.get];
+};
+
+module User = {
+  type t;
+  external bot : t => bool = "bot" [@@bs.get];
+};
+
+module Message = {
+  type t;
+  external content : t => string = "" [@@bs.get];
+  external channel : t => Channel.t = "" [@@bs.get];
+  external reply : t => string => unit = "" [@@bs.send];
```

Comment:
In `discord.js`, `Message.reply` returns promise. I think we should change this too now that BS 1.7 supports promise out of the box.